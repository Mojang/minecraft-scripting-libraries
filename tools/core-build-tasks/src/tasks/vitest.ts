import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
