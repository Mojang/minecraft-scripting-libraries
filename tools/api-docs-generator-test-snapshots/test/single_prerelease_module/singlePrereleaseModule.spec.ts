// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Single Pre-Release Module', () => {
    it('Properly generates documentation with a warning for a pre-release module when it is the only module', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['ts', 'msdocs'],
        });
    });
});
