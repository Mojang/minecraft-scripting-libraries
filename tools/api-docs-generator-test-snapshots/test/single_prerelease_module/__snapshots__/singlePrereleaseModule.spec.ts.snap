// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Single Pre-Release Module > Properly generates documentation with a warning for a pre-release module when it is the only module > # Of Files Generated 1`] = `4`;

exports[`Single Pre-Release Module > Properly generates documentation with a warning for a pre-release module when it is the only module > msdocs/scriptapi/TOC.yml 1`] = `
"  - name: prerelease-module
    items:
    - name: prerelease-module
      href: prerelease-module/prerelease-module.md
    - name: prerelease-module Changelog
      href: prerelease-module/changelog.md
"
`;

exports[`Single Pre-Release Module > Properly generates documentation with a warning for a pre-release module when it is the only module > msdocs/scriptapi/prerelease-module/changelog.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: prerelease-module Changelog
description: Changelog of the \`prerelease-module\` module
---
# \`prerelease-module\` Changelog

## Version Changes
- [1.0.0-beta](#100-beta)

## 1.0.0-beta
#### Added \`prerelease-module\` Module
"
`;

exports[`Single Pre-Release Module > Properly generates documentation with a warning for a pre-release module when it is the only module > msdocs/scriptapi/prerelease-module/prerelease-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: prerelease-module Module
description: Contents of the prerelease-module module
monikerRange: "=minecraft-bedrock-experimental"
---
# \`prerelease-module\` Module

> [!CAUTION]
> This module is still in pre-release.  It may change or it may be removed in future releases.

## [Changelog](changelog.md)

## Manifest Details
\`\`\`json
{
    "module_name": "prerelease-module",
    "version": "1.0.0-beta"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`1.0.0-beta\`
"
`;

exports[`Single Pre-Release Module > Properly generates documentation with a warning for a pre-release module when it is the only module > typescript/prerelease-module@1.0.0-beta.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @beta
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "prerelease-module",
 *   "version": "1.0.0-beta"
 * }
 * \`\`\`
 *
 */
"
`;
